src/formatter.rs

```diff
@@ -88,6 +89,32 @@ where
                 serializer.serialize_entry("level", level)?;
             }
 
+            #[cfg(feature = "ansi_logs")]
+            {
+                let level = match *metadata.level() {
+                    tracing::Level::ERROR => "error",
+                    tracing::Level::WARN => "warn",
+                    tracing::Level::INFO => "info",
+                    tracing::Level::DEBUG => "debug",
+                    tracing::Level::TRACE => "trace",
+                };
+
+                if self.with_level {
+                    let level_str = match level {
+                        "error" => nu_ansi_term::Color::Red,
+                        "warn" => nu_ansi_term::Color::Yellow,
+                        "info" => nu_ansi_term::Color::Green,
+                        "debug" => nu_ansi_term::Color::Blue,
+                        "trace" => nu_ansi_term::Color::Purple,
+                        _ => todo!("Level not supported"),
+                    }
+                    .bold()
+                    .paint(level);
+
+                    serializer.serialize_entry("level", &level_str.to_string())?;
```

Comment:
```suggestion
                    serializer.serialize_entry("level", level_str.as_str())?;
```

I think you can use as_str() here, then I think we'd save an allocation

Reply:
After trying a few things it seems like the `&level_str.to_string()` is the only way for the string to keep the ansi_codes in place.

Reply:
ah right, the as_str is just a reference to the "un painted" input. Thanks for checking!